### PR TITLE
Convert tcprecycle and arpcache modules to python

### PR DESCRIPTION
**Pull request by gregbdunn**
Move from using bash on these to python. Small bugfix in some of the wording in the tcprecycle instructions/documentation.

### CODE REVIEW

File: mod.d/arpcache.yaml [hunk 1]
```diff
@@ -16,41 +16,43 @@
...
+  from __future__ import print_function
+  import re
+  import subprocess
 
-  # read-in shared function
-  source functions.bash
+  try:
+      print("Determining if agressive ARP caching is is enabled")
+      arpstatus = subprocess.check_output(["sysctl", "net.ipv4.neigh.default.gc_thresh1"]).decode("utf-8")
+      if re.search("0", arpstatus):
```
Comment (Drudenhaus) [inline mod.d/arpcache.yaml]: The intent of looking for a zero is somewhat ambiguous. I recommend matching the whole string (it's fairly short) to show intent. e.g.

re.match(r"net\.ipv4\.neigh\.default\.gc_thresh1\ =\ 0", arpstatus)

File: mod.d/arpcache.yaml [hunk 2]
```diff
@@ -16,41 +16,43 @@
...
+          arp_success = True
+      else:
+          arp_success = False
+      if arp_success:
```
Comment (Drudenhaus) [inline mod.d/arpcache.yaml]: This if block seems redundant. The prints could be moved into the preceding if regex block.

File: mod.d/tcprecycle.yaml [hunk 3]
```diff
@@ -16,39 +16,42 @@
...
-          echo "[SUCCESS] Aggressive TCP recycling is disabled."
-  fi
-  echo "Please see https://github.com/awslabs/aws-ec2rescue-linux/blob/master/docs/modules/tcprecycle.md for further details"
+  from __future__ import print_function
+  import re
+  import subprocess
+
+  try:
+      print("Determining if TCP recycle is enabled")
+      recyclestatus = subprocess.check_output(["sysctl", "net.ipv4.tcp_tw_recycle"]).decode("utf-8")
```
Comment (Drudenhaus) [inline mod.d/tcprecycle.yaml]: Same thing as the other module regarding intent. Can match line with something like:
re.match(r"net\.ipv4\.tcp_tw_recycle\ =\ 0", recyclestatus)

File: mod.d/tcprecycle.yaml [hunk 4]
```diff
@@ -16,39 +16,42 @@
...
+      if re.search("0", recyclestatus):
+          recycle_success = True
+      else:
+          recycle_success = False
+      if recycle_success:
+          print("[SUCCESS] Aggressive TCP recycling is disabled.")
```
Comment (Drudenhaus) [inline mod.d/tcprecycle.yaml]: Same thing as in the other module regarding merging the if blocks.

File: mod.d/arpcache.yaml (hunk 2 above)
Comment (gregbdunn) [inline mod.d/arpcache.yaml]: I'd like to keep them separate for future splitting - e.g. remediation.

I'm okay with removing the redundancy now and just worrying about splitting it up further later if you feel strongly about it, though.

File: mod.d/arpcache.yaml (hunk 1 above)
Comment (gregbdunn) [inline mod.d/arpcache.yaml]: Can/will do.

File: mod.d/tcprecycle.yaml (hunk 3 above)
Comment (gregbdunn) [inline mod.d/tcprecycle.yaml]: Can/will do.

File: mod.d/tcprecycle.yaml (hunk 4 above)
Comment (gregbdunn) [inline mod.d/tcprecycle.yaml]: Same comment as above - I'd like them separate for easier modification later, but not strongly attached to the idea.

**Comment by gregbdunn** [comment]
I've updated the string matching so that this is more self explanatory.

Thoughts w/r/t keeping things split up and slightly redundant? I'm figuring if we add remediation in the future for these modules, the diagnostic function could return the _status variable and printing could be done elsewhere.

**Comment by gregbdunn** [comment]
Per offline discussion, removing the redundancy. We can always split it up again later if we add remediation support.

**Pull request merged**